rlw/shutdown-reason.ps1

```diff
@@ -1,12 +1,42 @@
-Write-Output '===== DECOMMISSIONING ====='
+# Since the decommision bundle runs before sending a shutdown to windows, we are unable to
+# obtain the reason from the OS as to why it is being shutdown. As a result, and to have
+# parity with Linux scripts, the os_decom_reason variable will be set to 'unavailable'.
+#
+# We pull the reason RightScale thinks the instance is going down and put it in
+# rs_decom_reason. Note that this variable will only be populated if we issue a
+# stop/terminate/reboot from either the RightScale dashboard or the API. It will
+# be blank if shutdown or reboot was initiated from the instance itself or via cloud provider's console.
+# Note we can't tell if a terminate was issued on a cloud provider's console, as we just know the system is going
+# down. rs_decom_reason possible values are:
+#   stop = instance is being stopped/shutdown but disk persists
+#   terminate = instance is being destroyed/deleted
+#   reboot = instance is being rebooted
+#   (blank) = instance not shutdown from RightScale dashboard or API
+#
+# DECOM_REASON is mimics rs_decom_reason. We export this value as an environment
+# parameter so subsequent scripts in the decommission bundle may have use it. The
+# following values are possible:
+#   stop
+#   terminate
+#   reboot
+#   unknown (default and when rs_decom_reason is blank)
 
-$systemStateEntry = Get-WinEvent -FilterHashtable @{logname='System'; id=1074} | Select -First 1
+Write-Output 'Decommissioning. Calculating reason for decommission: '
 
-switch -regex ($systemstateentry.message)
-{
-  '.*restart.*' { $reason = 'rebooting' }
-  '.*(power off|shutdown).*' { $reason = 'shutting down' }
-  default { $reason = 'unknown' }
+$os_decom_reason = 'unavailable'
+
+$RIGHTLINK_DIR = 'C:\Program Files\RightScale\RightLink'
+$rs_decom_reason = & "${RIGHTLINK_DIR}\rsc.exe" rl10 show /rll/proc/shutdown_kind
+
+if ($rs_decom_reason) {
+  $decom_reason = $rs_decom_reason
+} else {
+  $decom_reason = 'unknown'
```

Comment:
Few things:
* You can just remove all mentions of os_decom_reason since it'll never be available. I think the Linux/Windows versions can't and thus shouldn't mirror each other exactly. 
* Decom reason should always be set for a reboot/shutdown/terminate -- it wasn't set in the linux case of when the user initiated a reboot/shutdown, but that never happens on windows now
* If decom reason is NOT set, I believe it will only be for a service restart, which which case you'd export service_restart maybe. Well you'd have to try it out.